### PR TITLE
ci(NODE-5124): ignore curl 92 errors when downloading node

### PR DESCRIPTION
**Pull request by baileympearson**
### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/install-dependencies.sh [hunk 1]
```diff
@@ -74,7 +74,12 @@ echo "Node.js ${node_index_version} for ${operating_system}-${architecture} rele
 
 set -o xtrace
 
+set +o errexit
 curl "${CURL_FLAGS[@]}" "${node_download_url}" --output "$node_archive_path"
+code=$?
+if [ $code -ne 0 ] && [ $code -ne 92 ]; then echo "non-zero or 92 exit code while downloading NodeJS $code"; fi
```
Comment (nbbeeken) [inline .evergreen/install-dependencies.sh:80]:
```suggestion
if [[ $? -ne 0 && $? -ne 92 ]]; then echo "neither zero nor 92 exit code while downloading Node.js: got $code"; fi
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: .evergreen/install-dependencies.sh [hunk 2]
```diff
@@ -74,7 +74,11 @@ echo "Node.js ${node_index_version} for ${operating_system}-${architecture} rele
 
 set -o xtrace
 
+set +o errexit
 curl "${CURL_FLAGS[@]}" "${node_download_url}" --output "$node_archive_path"
+if [[ $? -ne 0 && $? -ne 92 ]]; then echo "neither zero nor 92 exit code while downloading Node.js: got $code"; fi
```
Comment (nbbeeken) [inline .evergreen/install-dependencies.sh:79]:
```suggestion
if [[ $? -ne 0 && $? -ne 92 ]]; then echo "neither zero nor 92 exit code while downloading Node.js: got $?"; fi
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Comment by durran** [comment]
Doesn't seem like even code 92 will work.

**Comment by nbbeeken** [comment]
The purple storm has subsided 🟣⛈️

**Pull request closed without merging**